overlord/snapstate/snapmgr.go

```diff
@@ -1111,3 +1096,27 @@ func (m *SnapManager) stopSnapServices(t *state.Task, _ *tomb.Tomb) error {
 
 	return err
 }
+
+func (m *SnapManager) cleanup(t *state.Task, _ *tomb.Tomb) error {
+	st := t.State()
+
+	st.Lock()
+	defer st.Unlock()
+
+	pb := &TaskProgressAdapter{task: t}
+	_, snapst, err := snapSetupAndState(t)
+	if err != nil {
+		pb.Notify(fmt.Sprintf("cannot clean up: %v", err))
```

Comment:
you can use logging function on the task directly no, instead of using a progressadapter ?